format.h

```diff
@@ -262,6 +216,72 @@ inline uint32_t clzll(uint64_t x) {
 # define FMT_ASSERT(condition, message) assert((condition) && message)
 #endif
 
+
+#if FMT_GCC_VERSION >= 400 || FMT_HAS_BUILTIN(__builtin_clz)
+# define FMT_BUILTIN_CLZ(n) __builtin_clz(n)
+#endif
+
+#if FMT_GCC_VERSION >= 400 || FMT_HAS_BUILTIN(__builtin_clzll)
+# define FMT_BUILTIN_CLZLL(n) __builtin_clzll(n)
+#endif
+
+#if defined(_MSC_VER)
+# include <intrin.h>  // _BitScanReverse, _BitScanReverse64
+
+namespace fmt {
+namespace internal {
+// Some compilers masquerade as both MSVC and GCC-likes or 
+// otherwise support __builtin_clz and __builtin_clzll, so
+// only define FMT_BUILTIN_CLZ using the MSVC intrinsics
+// if the clz and clzll builtins are not available.
+# if !defined(FMT_BUILTIN_CLZ)
```

Comment:
That's what I originally had, but I was being overly paranoid and worried about "what if there's a compiler with __builtin_clz support, but not __builtin_clzll, or the other way around?" and `#if defined(_MSC_VER) && (!defined(FMT_BUILTIN_CLZ) || !defined(FMT_BUILTIN_CLZLL))` was difficult for me to read and validate that it was correct.